src/vault-auth-chef/chefclient/path_auth.go

```diff
@@ -137,22 +139,42 @@ func (b *backend) verifyCreds(ctx context.Context, req *logical.Request, client,
 		return nil, errors.Wrap(err, "auth.test")
 	}
 
-	// Get node
-	node, err := c.Nodes.Get(client)
+	// Get data bags
+	// Iterate over configured data bags indexes and try to find the one for our client.
+	var dataBag interface{}
+	for _, dataBagPath := range config.DataBags {
+		dataBag, err = c.DataBags.GetItem(dataBagPath, client)
+		if err != nil {
+			b.logger.Debug(fmt.Sprintf("Looking for data bag in: %s", err.Error()))
+		}
+		if err == nil {
+			break
+		}
+	}
+	// Check if we realy got the data bag.
+	jsonData, err := json.Marshal(dataBag)
 	if err != nil {
-		b.logger.Warn(fmt.Sprintf("Chef auth error while get nodes: %s", err.Error()))
-		return nil, errors.Wrap(err, "nodes.list")
+		b.logger.Warn(fmt.Sprintf("Chef auth error while geting data bags: %s", err.Error()))
+		return nil, errors.Wrap(err, "data_bags.list")
 	}
+	dataBagMapRunList := gjson.GetBytes(jsonData, "run_list")
 	nodeRoles := make([]string, 0)
 	roleRe := regexp.MustCompile("^role\\[(.*)\\]$")
-	for _, role := range node.RunList {
+	for _, role := range dataBagMapRunList.Array() {
```

Comment:
It means that we should have run_list in data_bag only. But it calls manual changes while you change client's run list. I very don't like a lot of manual actions. May be we can add access to vault secrets by host groups in chef server? I think we should find some auth solution that can't be overwritten by client side, but can be dynamically changed on server side while we change node settings. Also I think it is possible to prevent override run_list from client side, so may be it is good to have multiply possible solutions...

Reply:
Maybe the fix is too much specific for my organization workflow. We keep run_list in data bags and manage roles to host assignment thru that. I don't think there is a possibility to prevent nodes run_list from overwriting - left some reading materials in https://github.com/svagner/vault-auth-chef/issues/6 on that topic. 

If you have any ideas on how to do proper and secure auth I would be glad to hear about them.